### PR TITLE
Avoid modifying inactive indexing fields after reading input streams

### PR DESCRIPTION
**Pull request by mgduda**
This PR addresses two sources of incorrect behavior when indexing fields (e.g., 
`cellsOnCell`) are deactivated in input streams through packages.

Firstly, indexing fields that were not read because they contained no active 
packages (i.e., they were deactivated) should not undergo global-to-local index
translation after their stream is read; doing so would apply the index
translation to either the default value for the indexing field in cases where
the field had never been set, or it would apply the index translation to what
are already local indices.

Secondly, indexing fields that were not read should not have their halos 
exchanged, since doing so would populate halo elements with local indices from
another block. More generally, there is no need to exchange halos after an input
stream is read for any fields that were deactivated in the input stream and
therefore not read.

### CODE REVIEW

**Review by dimomatt — APPROVED** [review]
Good to go, just curious about the capitalization conventions

File: src/framework/mpas_stream_manager.F [hunk 1]
```diff
@@ -4667,6 +4670,10 @@ subroutine exch_all_halos(allFields, streamFields, timeLevel, ierr) !{{{
         type (field2DInteger), pointer :: int2DField
         type (field3DInteger), pointer :: int3DField
 
+        character (len=StrKIND), pointer :: packages
+        logical :: active_field
```
[the remark this answers is not in the document]
Comment (mgduda) [reply src/framework/mpas_stream_manager.F:4674]: No reason. We use both styles throughout the infrastructure, and snake_case was probably just what was on my mind when I added this code. I can force-push an update to use snakeCase for consistency.

[the remark this answers is not in the document]
Comment (mgduda) [reply src/framework/mpas_stream_manager.F:4674]: After another look through the `mpas_stream_manager.F` code, it looks like the new code in this PR mirrors code, e.g., in the `build_streams` routine. A future PR should probably factor-out common code among various stream manager routines, and to make this repeated code easier to find, I'd propose keeping the snake_case code as-is for now.

[the remark this answers is not in the document]
Comment (dimomatt) [reply src/framework/mpas_stream_manager.F:4674]: yep, I think that makes sense. I don't really have an issue with snake case, especially when the rest of the filenames and modules follow that convention- was really just curious if there was a conscious change between the two styles at some point in the code.

**Review by dimomatt — APPROVED** [review]

**Pull request merged**